src/lib.rs

```diff
@@ -1,98 +1,53 @@
-use sha2::{Digest, Sha256};
-use std::sync::LazyLock;
-
-/// Length of a SHA256 hash in bytes.
-pub const HASH_LEN: usize = 32;
-
-/// Returns the digest of `input`.
-pub fn hash(input: &[u8]) -> Vec<u8> {
-    let mut hasher = Sha256::new();
-    hasher.update(input);
-    hasher.finalize().to_vec()
-}
+#![allow(dead_code)]
+#[cfg(feature = "poseidon2")]
+mod poseidon2;
+mod sha2;
```

Comment:
```suggestion
#[cfg(feature = "poseidon2")]
mod poseidon2;
#[cfg(feature = "sha")]
mod sha2;
```

Seems like `impl Poseidon2Hash { ... }` contains unused methods, so how about adding the directive above it?
Also, do we need to use `sha2` even in `feature = "poseidon2"`? It seems both features are mutually exclusive